python/lsst/jointcal/jointcal.py

```diff
@@ -1151,7 +1158,15 @@ def _iterate_fit(self, associations, fitter, max_steps, name, whatToFit,
 
                 break
             elif result == MinimizeResult.Chi2Increased:
-                self.log.warn("still some outliers but chi2 increases - retry")
+                self.log.warn("Still some outliers remaining but chi2 increased - retry")
+                # Check whether the increase was large enough to cause trouble.
+                chi2Ratio = chi2.chi2 / oldChi2.chi2
+                if chi2Ratio > 2:
+                    self.log.warn('Significant chi2 increase by a factor of %.4g / %.4g = %.4g',
+                                  chi2.chi2, oldChi2.chi2, chi2Ratio)
+                if chi2Ratio > self.config.failOnChi2Increase:
+                    raise ValueError('Large chi2 increase between steps: fit likely cannot converge.')
```

Comment:
Echoing above commentary on `config`, this smells an awful lot like an assertion, not a tunable parameter.